### ISSUE TITLE
Regression: slow mode changes

### ISSUE THREAD

**Issue by YaLTeR** [swaywm/sway#3771]
Starting from commit 7baaa3a0f80a28f58a3f95f7c3c832cb109b2aab (bisected with https://github.com/swaywm/wlroots/commit/b59370088e3290e2354ea01e30746a3711eea28f) mode changes are considerably slower.

Config: default with this appended:
```
set $first_monitor "Dell Inc. Dell S2716DG #ASOwvAqQj0Dd"
set $second_monitor "Unknown G-SYNC 241910 #ASNWvgsHCG3d"

bindsym $mod+u output $first_monitor  mode 2560x1440@119.998001Hz; \
               output $second_monitor mode 1920x1080@119.982002Hz

bindsym $mod+i output $first_monitor  mode 2560x1440@143.998001Hz; \
               output $second_monitor mode 1920x1080@144.001007Hz
```
Before that commit mode changes are really quick, maybe 0.3 seconds at most. Starting from that commit the displays take a good second to flash black.

**Comment by emersion** [comment]
Can you still reproduce?

**Comment by YaLTeR** [comment]
Looks like it was fixed in 1c329f2fe6ae344b3201447a586f6c9a571d44ff.
